### ISSUE TITLE
Validation stops working after reassigning state object

### ISSUE THREAD

**Issue by martinhla** [victorgarciaesgi/regle#24]
I have simple validation for an object `{ name: string }` where `name` is required. 

**Object Definition:**
```
const data = ref<{ name: string }>({
  name: '',
```
});

**Validation Rules:**
```
const { r$ } = useRegle(data, {
  name: {
    required,
  },
})
```

After reassigning the data object like this:

```
 data.value = {
    name: '',
  };
```
The validation stops working and remains in its state before the reassignment.

**Expected Behavior:** 
Validation should continue to work and reflect the current state of the data object after reassignment.

**Reproduction**
https://stackblitz.com/edit/vitejs-vite-jhebbf?file=src%2FApp.vue

**Comment by victorgarciaesgi** [comment]
Yep seems there is a problem when changing the whole object ref.

It works if mutating the direct value like `data.value.name = 'foo'`. 

I have a fix to to on the root array watch

**Comment by victorgarciaesgi** [comment]
Thanks for reporting all this bugs!

**Comment by victorgarciaesgi** [comment]
I will make it if you touch the object or nested object itself, and only the object (not nested properties) it calls `$touch` on the object

**Comment by victorgarciaesgi** [comment]
Fixed in `0.2.11`

**Comment by martinhla** [comment]
In the reproduction project, the fix helped. However, in my project, where I have a complex form, I encountered the same issue. I have tried to reproduce the issue in a smaller example with no luck so far. Maybe this will help: the validation state is `$invalid: true`, but all rules are `$invalid: false`.  `$errors` show message: **This list should have at least 3 items** but in `$value` object are 3 levels. Maybe I am missing something.

The validation only breaks when I reset the form object to its default values as the modal opens. If I remove this code, the validation works as expected:

```
watch(
  () => isOpen.value,
  async () => {
    if (isOpen.value) form.value = defaultModel();
  },
);

const defaultModel = () => {
  return {
    id: 0,
    name: "",
    localizations: [],
    levels: [],
  };
};
```

regle configuration:
```
const { r$ } = useRegle(
  form,
  () => ({
    levels: {
      minLength: minLength(3),
      maxLength: maxLength(10),
    },
  }),
);
```
regle state:
```
{
    "$dirty": true,
    "$anyDirty": true,
    "$invalid": true,
    "$valid": false,
    "$error": true,
    "$pending": false,
    "$errors": {
        "localizations": {
            "$errors": [],
            "$each": []
        },
        "levels": {
            "$errors": [
                "This list should have at least 3 items"
            ],
            "$each": []
        },
        "id": [],
        "name": []
    },
    "$silentErrors": {
        "localizations": {
            "$errors": [],
            "$each": []
        },
        "levels": {
            "$errors": [
                "This list should have at least 3 items",
                "This list should have maximum 10 items"
            ],
            "$each": []
        },
        "id": [],
        "name": []
    },
    "$ready": false,
    "$name": "root",
    "$fields": {
        "localizations": {
            "$field": {
                "$dirty": true,
                "$error": false,
                "$errors": [],
                "$valid": false,
                "$invalid": false,
                "$pending": false,
                "$silentErrors": [],
                "$anyDirty": true,
                "$ready": true,
                "$name": "localizations",
                "$value": [
                    {
                        "languageId": 1,
                        "name": "fsdfsdfsdf"
                    },
                    {
                        "languageId": 3,
                        "name": ""
                    }
                ],
                "$rules": {}
            },
            "$dirty": true,
            "$anyDirty": true,
            "$invalid": false,
            "$valid": false,
            "$error": false,
            "$pending": false,
            "$errors": {
                "$errors": [],
                "$each": [
                    {
                        "name": [],
                        "languageId": []
                    },
                    {
                        "name": [],
                        "languageId": []
                    }
                ]
            },
            "$silentErrors": {
                "$errors": [],
                "$each": [
                    {
                        "name": [
                            "This field is required"
                        ],
                        "languageId": []
                    },
                    {
                        "name": [
                            "This field is required"
                        ],
                        "languageId": []
                    }
                ]
            },
            "$ready": true,
            "$name": "localizations",
            "$each": [
                {
                    "$dirty": true,
                    "$anyDirty": true,
                    "$invalid": false,
                    "$valid": false,
                    "$error": false,
                    "$pending": false,
                    "$errors": {
                        "name": [],
                        "languageId": []
                    },
                    "$silentErrors": {
                        "name": [
                            "This field is required"
                        ],
                        "languageId": []
                    },
                    "$ready": true,
                    "$name": "localizations",
                    "$fields": {
                        "name": {
                            "$dirty": true,
                            "$error": false,
                            "$errors": [],
                            "$valid": true,
                            "$invalid": false,
                            "$pending": false,
                            "$silentErrors": [
                                "This field is required"
                            ],
                            "$anyDirty": true,
                            "$ready": true,
                            "$name": "name",
                            "$value": "fsdfsdfsdf",
                            "$rules": {
                                "required": {
                                    "$active": true,
                                    "$message": "This field is required",
                                    "$type": "required",
                                    "$params": [
                                        true
                                    ],
                                    "$path": "localizations.152159623.name.required",
                                    "$pending": false,
                                    "$valid": true,
                                    "$metadata": {},
                                    "_haveAsync": false,
                                    "$validating": false
                                }
                            }
                        },
                        "languageId": {
                            "$dirty": true,
                            "$error": false,
                            "$errors": [],
                            "$valid": false,
                            "$invalid": false,
                            "$pending": false,
                            "$silentErrors": [],
                            "$anyDirty": true,
                            "$ready": true,
                            "$name": "languageId",
                            "$value": 1,
                            "$rules": {}
                        }
                    },
                    "$value": {
                        "languageId": 1,
                        "name": "fsdfsdfsdf"
                    },
                    "$id": "152159623"
                },
                {
                    "$dirty": true,
                    "$anyDirty": true,
                    "$invalid": false,
                    "$valid": false,
                    "$error": false,
                    "$pending": false,
                    "$errors": {
                        "name": [],
                        "languageId": []
                    },
                    "$silentErrors": {
                        "name": [
                            "This field is required"
                        ],
                        "languageId": []
                    },
                    "$ready": true,
                    "$name": "localizations",
                    "$fields": {
                        "name": {
                            "$dirty": true,
                            "$error": false,
                            "$errors": [],
                            "$valid": false,
                            "$invalid": false,
                            "$pending": false,
                            "$silentErrors": [
                                "This field is required"
                            ],
                            "$anyDirty": true,
                            "$ready": true,
                            "$name": "name",
                            "$value": "",
                            "$rules": {
                                "required": {
                                    "$active": false,
                                    "$message": "This field is required",
                                    "$type": "required",
                                    "$params": [
                                        false
                                    ],
                                    "$path": "localizations.3162164051.name.required",
                                    "$pending": false,
                                    "$valid": true,
                                    "$metadata": {},
                                    "_haveAsync": false,
                                    "$validating": false
                                }
                            }
                        },
                        "languageId": {
                            "$dirty": true,
                            "$error": false,
                            "$errors": [],
                            "$valid": false,
                            "$invalid": false,
                            "$pending": false,
                            "$silentErrors": [],
                            "$anyDirty": true,
                            "$ready": true,
                            "$name": "languageId",
                            "$value": 3,
                            "$rules": {}
                        }
                    },
                    "$value": {
                        "languageId": 3,
                        "name": ""
                    },
                    "$id": "3162164051"
                }
            ],
            "$value": [
                {
                    "languageId": 1,
                    "name": "fsdfsdfsdf"
                },
                {
                    "languageId": 3,
                    "name": ""
                }
            ]
        },
        "levels": {
            "$field": {
                "$dirty": true,
                "$error": false,
                "$errors": [],
                "$valid": true,
                "$invalid": false,
                "$pending": false,
                "$silentErrors": [
                    "This list should have at least 3 items",
                    "This list should have maximum 10 items"
                ],
                "$anyDirty": true,
                "$ready": true,
                "$name": "levels",
                "$value": [
                    {
                        "id": -1,
                        "name": "",
                        "description": "",
                        "level": 1,
                        "localizations": [
                            {
                                "id": 0,
                                "languageId": 3,
                                "name": "",
                                "description": ""
                            },
                            {
                                "id": 0,
                                "languageId": 1,
                                "name": "sdf",
                                "description": ""
                            }
                        ]
                    },
                    {
                        "id": -2,
                        "name": "",
                        "description": "",
                        "level": 2,
                        "localizations": [
                            {
                                "id": 0,
                                "languageId": 3,
                                "name": "",
                                "description": ""
                            },
                            {
                                "id": 0,
                                "languageId": 1,
                                "name": "sdf",
                                "description": ""
                            }
                        ]
                    },
                    {
                        "id": -3,
                        "name": "",
                        "description": "",
                        "level": 3,
                        "localizations": [
                            {
                                "id": 0,
                                "languageId": 3,
                                "name": "",
                                "description": ""
                            },
                            {
                                "id": 0,
                                "languageId": 1,
                                "name": "sdf",
                                "description": ""
                            }
                        ]
                    }
                ],
                "$rules": {
                    "minLength": {
                        "$active": true,
                        "$message": "This list should have at least 3 items",
                        "$type": "minLength",
                        "$params": [
                            3
                        ],
                        "$path": "levels.minLength",
                        "$pending": false,
                        "$valid": true,
                        "$metadata": {},
                        "_haveAsync": false,
                        "$validating": false
                    },
                    "maxLength": {
                        "$active": true,
                        "$message": "This list should have maximum 10 items",
                        "$type": "maxLength",
                        "$params": [
                            10
                        ],
                        "$path": "levels.maxLength",
                        "$pending": false,
                        "$valid": true,
                        "$metadata": {},
                        "_haveAsync": false,
                        "$validating": false
                    }
                }
            },
            "$dirty": true,
            "$anyDirty": true,
            "$invalid": false,
            "$valid": true,
            "$error": false,
            "$pending": false,
            "$errors": {
                "$errors": [],
                "$each": []
            },
            "$silentErrors": {
                "$errors": [
                    "This list should have at least 3 items",
                    "This list should have maximum 10 items"
                ],
                "$each": []
            },
            "$ready": true,
            "$name": "levels",
            "$each": [],
            "$value": [
                {
                    "id": -1,
                    "name": "",
                    "description": "",
                    "level": 1,
                    "localizations": [
                        {
                            "id": 0,
                            "languageId": 3,
                            "name": "",
                            "description": ""
                        },
                        {
                            "id": 0,
                            "languageId": 1,
                            "name": "sdf",
                            "description": ""
                        }
                    ]
                },
                {
                    "id": -2,
                    "name": "",
                    "description": "",
                    "level": 2,
                    "localizations": [
                        {
                            "id": 0,
                            "languageId": 3,
                            "name": "",
                            "description": ""
                        },
                        {
                            "id": 0,
                            "languageId": 1,
                            "name": "sdf",
                            "description": ""
                        }
                    ]
                },
                {
                    "id": -3,
                    "name": "",
                    "description": "",
                    "level": 3,
                    "localizations": [
                        {
                            "id": 0,
                            "languageId": 3,
                            "name": "",
                            "description": ""
                        },
                        {
                            "id": 0,
                            "languageId": 1,
                            "name": "sdf",
                            "description": ""
                        }
                    ]
                }
            ]
        },
        "id": {
            "$dirty": true,
            "$error": false,
            "$errors": [],
            "$valid": false,
            "$invalid": false,
            "$pending": false,
            "$silentErrors": [],
            "$anyDirty": true,
            "$ready": true,
            "$name": "id",
            "$value": 0,
            "$rules": {}
        },
        "name": {
            "$dirty": true,
            "$error": false,
            "$errors": [],
            "$valid": false,
            "$invalid": false,
            "$pending": false,
            "$silentErrors": [],
            "$anyDirty": true,
            "$ready": true,
            "$name": "name",
            "$value": "",
            "$rules": {}
        }
    },
    "$value": {
        "id": 0,
        "name": "",
        "localizations": [
            {
                "languageId": 1,
                "name": "fsdfsdfsdf"
            },
            {
                "languageId": 3,
                "name": ""
            }
        ],
        "levels": [
            {
                "id": -1,
                "name": "",
                "description": "",
                "level": 1,
                "localizations": [
                    {
                        "id": 0,
                        "languageId": 3,
                        "name": "",
                        "description": ""
                    },
                    {
                        "id": 0,
                        "languageId": 1,
                        "name": "sdf",
                        "description": ""
                    }
                ]
            },
            {
                "id": -2,
                "name": "",
                "description": "",
                "level": 2,
                "localizations": [
                    {
                        "id": 0,
                        "languageId": 3,
                        "name": "",
                        "description": ""
                    },
                    {
                        "id": 0,
                        "languageId": 1,
                        "name": "sdf",
                        "description": ""
                    }
                ]
            },
            {
                "id": -3,
                "name": "",
                "description": "",
                "level": 3,
                "localizations": [
                    {
                        "id": 0,
                        "languageId": 3,
                        "name": "",
                        "description": ""
                    },
                    {
                        "id": 0,
                        "languageId": 1,
                        "name": "sdf",
                        "description": ""
                    }
                ]
            }
        ]
    }
}
```

**Comment by victorgarciaesgi** [comment]
Couldn't reproduce with what you described :( 
Do you have a step by step repro of your state and what action you do to your form?

**Comment by victorgarciaesgi** [comment]
I think I know, to explain here is how collection fields works.

Your have the main field with properties like `$dirty` ...etc, but the root ones are the aggregations of all the `$each` rules
(in red).
In blue is the `$field`, which regroup the rules concerning the Array **itself**, not each element.
<img width="450" alt="image" src="https://github.com/user-attachments/assets/0a906eba-07ea-48c6-b741-95b2ca0222ff">

When doing nested collections, I think i'm only collecting the `$each` section errors

**Comment by victorgarciaesgi** [comment]
Actually I tested many cases, but couldn't reproduce your output :( 
It worked fine on my side

**Comment by victorgarciaesgi** [comment]
Your `r$` schema your pasted is definitely wrong, but I just need to know how to get there

**Comment by martinhla** [comment]
I finally figured out why validation was still broken in my project. After creating a new package lock, I noticed that upgrading Vue to version `3.5.13` resolved the issue. The problem was with the previous version, `3.4.38`.

there is example: https://stackblitz.com/edit/vitejs-vite-jhebbf?file=src%2FApp.vue

**Comment by victorgarciaesgi** [comment]
Thanks your for the reproduction! I hope the problem is not on Vue side

**Comment by victorgarciaesgi** [comment]
Found the issue, not solid clear why this was not working on `3.4.x`. But 3.5 was not having this bug!
Fixed in `0.2.13`

**Comment by martinhla** [comment]
Works, great work!
